### PR TITLE
[NEW] enable advanced SMTP config

### PR DESCRIPTION
**Pull request by njeudy**
- enable advanced SMTP config in odoo.conf

Needed by #107

### CODE REVIEW

**Comment by njeudy** [comment]
@Yajo some did did'nt pass but It's not due to my commit .. what should I do ?

File: conf.d/40-smtp.conf [hunk 1]
```diff
@@ -1,2 +1,7 @@
 ; Prepared to link with https://hub.docker.com/r/tecnativa/postfix-relay
 smtp_server = $SMTP_SERVER
+smtp_port = $SMTP_PORT
+smtp_user = $SMTP_USER
+smtp_password = $SMTP_PASSWORD
+smtp_ssl = $SMTP_SSL
+email_from = $EMAIL_FROM
```
Comment (Yajo) [inline conf.d/40-smtp.conf]: 🤔 Does this actually work? Isn't a record in `ir.mail.server` required?

cc   @pedrobaeza

**Comment by Yajo** [comment]
Yeah, Travis is sick since recently with #109, dunno why, it started happening suddenly. I'll have to do manual tests instead.

Please add docs under development (opening port for mailhog) and testing (using extra routing path from traefik).

File: conf.d/40-smtp.conf (hunk 1 above)
Comment (njeudy) [reply conf.d/40-smtp.conf]: I will check in my test database .. but think I have no ir.mail.server

Comment (njeudy) [reply conf.d/40-smtp.conf]: I have to check this 4 params I just take odoo.conf from https://github.com/odoo/docker/blob/16d5988408434e4678f3bf6318aabad542dec4ee/11.0/odoo.conf.

server and port is working but never really try this 4 params.

**Comment by Yajo** [comment]
Please rebase and add tests, now that #109 is fixed

**Comment by njeudy** [comment]
@yago: I have to correct travis or this is in your work list ?

**Comment by Yago** [comment]
@njeudy Not sure, ask @Yajo just in case 😅

**Comment by njeudy** [comment]
@Yago  oups :) sorry @Yajo  I have to correct travis or this is in your work list ?

**Comment by njeudy** [comment]
@Yajo All green now :) is it ok for you ?

File: 8.0.Dockerfile [hunk 2]
```diff
@@ -20,7 +20,13 @@ ONBUILD ARG COMPILE=true
 ONBUILD ARG CONFIG_BUILD=true
 ONBUILD ARG PIP_INSTALL_ODOO=true
 ONBUILD ARG ADMIN_PASSWORD=admin
+
```
Comment (Yajo) [inline 8.0.Dockerfile]: :fire:

File: 11.0.Dockerfile [hunk 3]
```diff
@@ -21,6 +21,12 @@ ONBUILD ARG CONFIG_BUILD=true
 ONBUILD ARG PIP_INSTALL_ODOO=true
 ONBUILD ARG ADMIN_PASSWORD=admin
 ONBUILD ARG SMTP_SERVER=smtp
+ONBUILD ARG SMTP_PORT=25
+ONBUILD ARG SMTP_USER=false
+ONBUILD ARG SMTP_PASSWORD=false
+ONBUILD ARG SMTP_SSL=false
+ONBUILD ARG EMAIL_FROM=""
+
```
Comment (Yajo) [inline 11.0.Dockerfile]: 🔥

**Comment by njeudy** [comment]
@Yajo ok done :) sorry miss this !

File: 11.0.Dockerfile [hunk 4]
```diff
@@ -36,6 +41,11 @@ ONBUILD ENV ADMIN_PASSWORD="$ADMIN_PASSWORD" \
             PGDATABASE="$PGDATABASE" \
             PROXY_MODE="$PROXY_MODE" \
             SMTP_SERVER="$SMTP_SERVER" \
+            SMTP_PORT=$SMTP_PORT \
```
Comment (Yajo) [inline 11.0.Dockerfile]: Is there any reason why this has no surrounding quotes? I'd put them, just for consistency.

File: 8.0.Dockerfile [hunk 5]
```diff
@@ -38,6 +43,11 @@ ONBUILD ENV ADMIN_PASSWORD="$ADMIN_PASSWORD" \
             PGDATABASE="$PGDATABASE" \
             PROXY_MODE="$PROXY_MODE" \
             SMTP_SERVER="$SMTP_SERVER" \
+            SMTP_PORT=$SMTP_PORT \
```
Comment (Yajo) [inline 8.0.Dockerfile]: Same about quotes.

**Comment by Yajo** [comment]
Apart from those 2 minor details, there's still one thing remaining here: tests.

You're adding here the default settings to make work the bundled `smtp` relay, which is really cool! I guess we should now create an odoo shell script that sends an email with those default settings and checks the relay returns an OK.

Wanna try yourself? 😉

**Comment by njeudy** [comment]
@Yajo yes I will give it a try, but can you merge this one (will open a new for test) because today I need to rebuild my personnal docker all the time there is a change just to add this params.

It's up to you, i will work on test next week I think I have some module to end before :)

**Comment by Yajo** [comment]
Fine, we can wait. I really like the feature, but Doodba is really a very critical piece of software in many companies' production toolchain, I usually don't merge anything without tests, not even my own PRs... Surely you understand 😇

**Comment by Yajo** [comment]
I added the test, it was quite simple.

**Comment by njeudy** [comment]
@Yajo try to make ci green but did not understand actual error .. can you help me ?

**Comment by Yajo** [comment]
It seems the DB was missing, I did a new try.

**Comment by Yajo** [comment]
I added a new test scaffolding. The dotd one is a little bit crowded and did not require a DB until now. Adding a DB into it clutters it even more, so I guess it's time to have a new one for the settings things. It will help testing #62 too.

**Comment by Yajo** [comment]
Finally I decided to skip the mail sending part. It's too low level to investigate. Just testing settings are OK should be OK for this project's support surface. Thanks pal!

**Pull request merged**